README.md

```diff
@@ -66,11 +66,106 @@ Select desired `Start` and `End` waypoints using the `RMF Panel` and click the `
 ```bash
 source ~/rmf_demos_ws/install/setup.bash
 ros2 launch demos office_loop.launch.xml
-``` 
+```
 
 ![](docs/media/loop_request.gif)
 
----
+#### Secure Ros2 Office Demo
+
+The office demo can be run in secure mode using the [ROS 2 DDS-Security integration](https://design.ros2.org/articles/ros2_dds_security.html) (SROS2) capabilities. This security features will provide encryption, authentication and access control to the whole ROS2 setup of RMF.
+
+To use ROS2 to secure the office demo a few environmental variables need to be set. Adding them to a script would make it easier to source them in any shell where we need to run ros2 secured nodes.
+
+```bash
+echo 'export ROS_SECURITY_KEYSTORE=~/rmf_demos_ws/keystore
+export ROS_SECURITY_ENABLE=true
+export ROS_SECURITY_STRATEGY=Enforce
+export ROS_DOMAIN_ID=42' > sros2_environment.sh
+```
+
+A few security artifacts like signed permission files, identity certificates and Certificate Authorities (CA) are needed to enable the DDS Security. They can be generated using the security tools from the ROS2 cli. Because of [#242](https://github.com/ros2/sros2/issues/242) and until its solution [#238](https://github.com/ros2/sros2/pull/238) is released it is recommended to generate the security artifacts before switching to `cyclone_dds`.
+
+```
+source sros2_environment.sh
+mkdir keystore
+ros2 security generate_artifacts -k keystore -p ./install/demos/share/demos/sros2/policies/office.policy.xml
+```
+
+It is recommended to use `cyclone dds` with the secure version of the demo, make sure it is properly installed, with security support and selected through the correspondent environment variable ([here](https://index.ros.org/doc/ros2/Installation/DDS-Implementations/Working-with-Eclipse-CycloneDDS/) you can find instructions on how to do it). Adding this to the environment script makes it easier to source in any terminal that might need it,
+
+```bash
+echo 'export RMW_IMPLEMENTATION=rmw_cyclonedds_cpp' >> sros2_environment.sh
+```
+
+Because SROS2 does not support `ros2 launch` yet all the different nodes will have to be launched individually. A `yaml` file with the parameters of all the nodes is provided. One terminal per execution is recommended for an easier debug of the system. Make sure you `source /opt/ros/foxy/setup.bash` and `source ~/rmf_demos_ws/install/setup.bash` in each of them before executing the different commands.
```

Comment:
Sourcing `~/rmf_demos_ws/install/setup.bash` will automatically source the ROS distro used to build the workspace as an underlay.

Reply:
```suggestion
Because SROS2 does not support `ros2 launch` yet all the different nodes will have to be launched individually. A `yaml` file with the parameters of all the nodes is provided. One terminal per execution is recommended for an easier debug of the system. Make sure you `source ~/rmf_demos_ws/install/setup.bash` in each of them before executing the different commands.
```